game/renderware_binary_stream.ksy

```diff
@@ -146,6 +146,17 @@ types:
         value: format & 0x00000080 != 0
       is_native:
         value: format & 0x01000000 != 0
+      num_uv_layers_raw:
+        value: (format & 0x00ff_0000) >> 16
+      num_uv_layers:
+        value: |
+          num_uv_layers_raw == 0
+            ? is_textured
+              ? 1
+              : is_textured2
+                ? 2
+                : 0
+            : num_uv_layers_raw
```

Comment:
I think that `is_textured2` should take precedence over `is_textured`.

Although I haven't seen a file with `num_uv_layers_raw == 0`, I have found one that has a geometry with `is_textured2` here: https://libertycity.net/files/gta-san-andreas/188330-fog-endsieg.html

The relevant file is called `hunter.dff` (I can probably upload it here for convenience: [**hunter.dff.tgz**](https://github.com/kaitai-io/kaitai_struct_formats/files/11181908/hunter.dff.tgz)). As you can see, the geometries with `"num_uv_layers": 2` have both `is_textured2` and `is_textured` set (not only `is_textured2`):

<pre><code>                  "is_textured": true,
                  "is_textured2": <ins><strong>true</strong></ins>,
                  "num_uv_layers": 2,
                  "num_uv_layers_raw": 2,

                  "is_textured": true,
                  "is_textured2": false,
                  "num_uv_layers": 1,
                  "num_uv_layers_raw": 1,

                  "is_textured": true,
                  "is_textured2": <ins><strong>true</strong></ins>,
                  "num_uv_layers": 2,
                  "num_uv_layers_raw": 2,

                  "is_textured": true,
                  "is_textured2": <ins><strong>true</strong></ins>,
                  "num_uv_layers": 2,
                  "num_uv_layers_raw": 2,

                  "is_textured": true,
                  "is_textured2": false,
                  "num_uv_layers": 1,
                  "num_uv_layers_raw": 1,

                  ...</code></pre>

For this specific file, it's not a problem because `num_uv_layers_raw` is correctly set, but we can see that just because `is_textured` is set doesn't mean the number of layers is `1`. So you should check `is_textured2` first - if it's true, then the number of layers is `2`. Only if it's false and `is_textured` is true, then there's probably just 1 layer.

This interpretation also matches some source codes I randomly came across (by searching [for `rpGEOMETRYTEXTURED2`](https://github.com/search?q=rpGEOMETRYTEXTURED2&type=code) on GitHub):

https://github.com/atirut-w/gta3-godot/blob/027f2ccd537f4cdedd969f667575bd195c7640e8/scripts/renderware/rw_geometry.gd#L98-L99

```gd
			if format & rpGEOMETRYTEXTURED2:
				uv_count = 2
```

https://github.com/Sor3nt/manhunt-studio/blob/b145349d42dd25df3cb61092a4ee37aa85d88ebb/src/Plugin/Loader/Renderware/Chunk/Geometry.js#L232-L249

```js
            if ((formatFlags & Renderware.rpGEOMETRYTEXTURED) === Renderware.rpGEOMETRYTEXTURED || (formatFlags & Renderware.rpGEOMETRYTEXTURED2) === Renderware.rpGEOMETRYTEXTURED2){

                for(let i = 0; i < this.rootData.vertexCount; i++){
                    this.result.uv1.push([
                        struct.binary.float32(),
                        struct.binary.float32()
                    ]);
                }
            }


            if ((formatFlags & Renderware.rpGEOMETRYTEXTURED2) === Renderware.rpGEOMETRYTEXTURED2){
                for(let i = 0; i < this.rootData.vertexCount; i++){
                    this.result.uv2.push([
                        struct.binary.float32(),
                        struct.binary.float32()
                    ]);
                }
            }
```

Reply:
Yeah, after seeing all this, I agree, I comitted a change that does not give precedence to `is_textured2` over `is_textured`, thanks for all the research!